### PR TITLE
Fix Issue 19202 - deprecated eponymous template prints no warning

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

**Comment by WalterBright** [comment]
@RazvanN7 something's up with appveyor

**Comment by RazvanN7** [comment]
@WalterBright It's unrelated to this PR. It always appears when you target stable. I think that @TurkeyMan 's C++ namespace PR wasn't merged in stable.

File: test/fail_compilation/fail19202.d [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// REQUIRED_ARGS: -de
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail19202.d(10): Deprecation: variable `fail19202.X!().X` is deprecated
+---
+*/
+
+void main() {
```
[the remark this answers is not in the document]
Comment (RazvanN7) [reply test/fail_compilation/fail19202.d]: Done.

**Comment by thewilsonator** [comment]
```
Test fail_compilation/fail19202.d failed: 
expected:
----
fail_compilation/fail19202.d(10): Deprecation: variable `fail19202.X!().X` is deprecated
----
actual:
----
fail_compilation/fail19202.d(11): Deprecation: variable `fail19202.X!().X` is deprecated
```

**Comment by RazvanN7** [comment]
@thewilsonator yeah, forgot to update the line number. Should be good now

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**